streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -138,7 +138,7 @@ synchronized boolean lock(final TaskId taskId) throws IOException {
         }
 
         try {
-            lockFile = new File(directoryForTask(taskId), LOCK_FILE_NAME);
+            lockFile = new File(stateDir, taskId + LOCK_FILE_NAME);
```

Comment:
why remove the `directoryForTask` method?

Reply:
The location of the lock file is lifted one level up.

Reply:
After discussion on #4713 I think this idea should actually work. Nit: Can we rename the lock to `LOCK_FILE_NAME + " -" + taskId` though.

Reply:
I can rename the lock if people think we can continue with this solution.